src/test/java/de/tum/in/www1/artemis/util/GitUtilService.java

```diff
@@ -30,7 +30,7 @@ public class GitUtilService {
     // Note: the first string has to be same as artemis.repo-clone-path (see src/test/resources/config/application-artemis.yml) because here local git repos will be cloned
     private final Path localPath = Path.of(".", "repos", "server-integration-test").resolve("test-repository").normalize();
 
-    private final Path remotePath = Files.createTempDirectory("remotegittest").resolve("scm/test-repository");
+    private final Path remotePath = Path.of(System.getProperty("java.io.tmpdir")).resolve("remotegittest/scm/test-repository");
```

Comment:
Just a question (the change is fine by me): Why is this change necessary? Shouldn't it work with the code before as well?

Reply:
`createTempDirectory` creates a new temporary directory every time this service gets instantiated (i.e. every time a local test run gets started) even if no methods on this service are called. This means that a new folder gets created, but never deleted. Not we just save the path here and create the folder only when calling `initRepo`, which means that the folder also gets deleted when later calling `deleteRepos`